src/client.ts

```diff
@@ -6881,7 +6881,17 @@ export class MatrixClient extends EventEmitter {
      * @param {module:client.callback} callback Optional.
      * @return {Promise} Resolves: On success, the empty object
      */
-    public logout(callback?: Callback): Promise<{}> {
+    public async logout(callback?: Callback): Promise<{}> {
+        if (this.crypto?.backupManager?.getKeyBackupEnabled()) {
+            try {
+                while (await this.crypto.backupManager.backupPendingKeys(200) > 0);
+            } catch (err) {
+                logger.log(
```

Comment:
i believe this could/should be an `error`?

Reply:
Yeah, that sounds likely.